### PR TITLE
First iteration of SQL DB deployment

### PR DESCRIPTION
**Pull request by smartguest**
Resource Deployment now allows for deployment of a simple database on an existing server and resource group along with the creation of a new firewall rule (part of the original notebook for creating databases).

Elastic pool and database servers will open up a link to create those (as noted on the deployment options page). 

NOTE: Commented out code and the azure-sql-db-old.ipynb notebook are part of an an earlier attempt to use Azure Virtual Network Public IP addresses for the firewall rule (firewall rule needs IVP4 formatted IP addresses to work). They will be removed once deemed unnecessary or merged back in when there is a way to retrieve IPV4 formatted addresses from a REST call.

### CODE REVIEW

File: extensions/resource-deployment/notebooks/azuredb/create-sqldb-old.ipynb [hunk 1]
```diff
@@ -0,0 +1,218 @@
+{
```
Comment (kburtram) [inline extensions/resource-deployment/notebooks/azuredb/create-sqldb-old.ipynb:1]: filename say `old`.  Should this file be included in the PR?

File: extensions/resource-deployment/notebooks/azuredb/create-sqldb.ipynb [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    "nbformat": 4,
+    "cells": [
+        {
+            "cell_type": "markdown",
+            "source": [
+                "Create Azure SQL Database\n",
+                "============================================\n",
+                "\n",
+                "Steps of this procedure include:\n",
+                "1. Connect to Azure subscription\n",
```
Comment (kburtram) [inline extensions/resource-deployment/notebooks/azuredb/create-sqldb.ipynb:30]: all the steps say `1.`

File: extensions/resource-deployment/notebooks/azuredb/create-sqldb.ipynb [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                "    if output:\n",
+                "        output = output.decode(\"utf-8\")\n",
+                "        if printOutput:\n",
+                "            print(output)\n",
+                "        try:\n",
+                "            return json.loads(output)\n",
+                "        except:\n",
+                "            return output\r\n",
+                "\r\n",
+                "def get_random_string(length):\r\n",
```
Comment (kburtram) [inline extensions/resource-deployment/notebooks/azuredb/create-sqldb.ipynb:78]: I couldn't see where this method is being called from

File: extensions/resource-deployment/notebooks/azurevm/create-sqlvm.ipynb [hunk 4]
```diff
@@ -0,0 +1,396 @@
+{
```
Comment (kburtram) [inline extensions/resource-deployment/notebooks/azurevm/create-sqlvm.ipynb:1]: the `SQL VM` deployment notebook is also part of this PR?

File: extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/constants.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+export const AzureAccountDropdownLabel = localize('deployAzureSQLDB.AzureAccountDropdownLabel', "Azure Account");
+export const AzureAccountSubscriptionDropdownLabel = localize('deployAzureSQLDB.AzureSubscriptionDropdownLabel', "Subscription");
+export const AzureAccountDatabaseServersDropdownLabel = localize('deployAzureSQLDB.AzureDatabaseServersDropdownLabel', "Server");
+export const AzureAccountResourceGroupDropdownLabel = localize('deployAzureSQLDB.ResourceGroup', "Resource Group");
+//export const AzureAccountRegionDropdownLabel = localize('deployAzureSQLDB.AzureRegionDropdownLabel', "Region (for Public IP Address)");
+
+// Database settings page constants
+export const DatabaseSettingsPageTitle = localize('deployAzureSQLDB.DatabaseSettingsPageTitle', "Database settings");
+export const DatabaseSettingsPageDescription = localize('deployAzureSQLDB.DatabaseSettingsPageDescription', "");
+//export const PublicIPDropdownLabel = localize('deployAzureSQLDB.PublicIPDropdownLabel', "Public IP");
```
Comment (kburtram) [inline extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/constants.ts:28]: dead code

File: extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/pages/databaseSettingsPage.ts [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+		});
+	}
+
+	public onLeave(): void {
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+			return true;
+		});
+	}
+
+	// private async createPublicIPDropdown(view: azdata.ModelView) {
```
Comment (kburtram) [inline extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/pages/databaseSettingsPage.ts:99]: lots of dead code in this PR.  If we want to commit this code like this, it's good practice to add a comment why the code is disabled and a name\date of who's adding the deadcode.  that helps when trying to clean-up these comment blocks in the future.

File: extensions/resource-deployment/notebooks/azuredb/create-sqldb.ipynb (hunk 3 above)
Comment (smartguest) [reply extensions/resource-deployment/notebooks/azuredb/create-sqldb.ipynb:78]: This was also in Aasim's SQLVM notebook and it wasn't used there either, it is kept here for consistency, unless it should be removed.

File: extensions/resource-deployment/notebooks/azurevm/create-sqlvm.ipynb (hunk 4 above)
Comment (smartguest) [reply extensions/resource-deployment/notebooks/azurevm/create-sqlvm.ipynb:1]: This is from Aasim's branch, it shouldn't be reviewed

**Pull request closed without merging**